Okta/OktaOidc/Internal/Tasks/macOS/OktaOidcBrowserTaskMAC.swift

```diff
@@ -0,0 +1,94 @@
+        if let redirectServerConfiguration = redirectServerConfiguration {
+            redirectServer = OktaRedirectServer(successURL: redirectServerConfiguration.successRedirectURL,
+                                                port: redirectServerConfiguration.port ?? 0)
+        }
+
+        super.init(config: config, oktaAPI: oktaAPI)
+
+        NSAppleEventManager.shared().setEventHandler(self,
+                                                     andSelector: #selector(OktaOidcBrowserTaskMAC.handleEvent(_:withReplyEvent:)),
+                                                     forEventClass: AEEventClass(kInternetEventClass),
+                                                     andEventID: AEEventID(kAEGetURL))
+    }
+
+    deinit {
+        redirectServer?.stopListener()
+        NSAppleEventManager.shared().removeEventHandler(forEventClass: AEEventClass(kInternetEventClass), andEventID: AEEventID(kAEGetURL))
+    }
+
+    override var userAgentSession: OIDExternalUserAgentSession? {
+        didSet {
+            self.redirectServer?.redirectHandler.currentAuthorizationFlow = userAgentSession
+        }
+    } 
+
+    override func signIn(callback: @escaping ((OIDAuthState?, OktaOidcError?) -> Void)) {
+        if let redirectServer = self.redirectServer {
+            do {
+                redirectUrl = try redirectServer.startListener()
+            } catch(let error) {
+                callback(nil, OktaOidcError.redirectServerError("Redirect server error: \(error.localizedDescription)"))
+                return
+            }
+        } else {
+            redirectUrl = self.config.redirectUri
+        }
+
+        super.signIn(callback: callback)
+    }
+
+    override func signOutWithIdToken(idToken: String, callback: @escaping (Void?, OktaOidcError?) -> Void) {
+        if let redirectServer = self.redirectServer {
+            do {
+                redirectUrl = try redirectServer.startListener()
+            } catch(let error) {
+                callback(nil, OktaOidcError.redirectServerError("Redirect server error: \(error.localizedDescription)"))
+                return
+            }
+        } else {
+            redirectUrl = self.config.logoutRedirectUri
+        }
+
+        super.signOutWithIdToken(idToken: idToken, callback: callback)
+    }
+
+    override func signInRedirectUri() -> URL? {
+        return redirectUrl
+    }
+
+    override func signOutRedirectUri() -> URL? {
+        return redirectUrl
+    }
+
+    override func externalUserAgent() -> OIDExternalUserAgent? {
+        return OIDExternalUserAgentMac()
+    }
+    
+    @objc func handleEvent(_ event: NSAppleEventDescriptor!, withReplyEvent: NSAppleEventDescriptor!) {
+        let eventDescriptor = event.paramDescriptor(forKeyword: AEEventID(keyDirectObject))
+        if let stringValue = eventDescriptor?.stringValue,
+           let url = URL(string: stringValue) {
+            self.resume(with: url)
```

Comment:
style-nit: Need extra indentation here

Reply:
Not sure that it is required. There are 4 spaces offset relative to `if`

Reply:
Ah - the newline-d `let` threw me off here. Thanks for the clarification!